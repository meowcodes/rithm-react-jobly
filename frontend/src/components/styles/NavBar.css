.NavBar {
  position        : relative;
  text-align      : right;
  background-color: rgba(161, 218, 213, 0.425);
  font-size       : 1rem;
  height          : 2.5rem;
}

#NavBar-logo{
  left    : 1em;
  position: absolute;
}

.NavBar section {
  display     : inline-block;
  margin-right: 1rem;
}

.NavBar a {
  display : inline-block;
  position: relative;
  padding : 0.5rem 0.5rem;
}

a.active {
  font-weight: bold;
}