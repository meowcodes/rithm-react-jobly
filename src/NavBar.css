nav a.active {
    font-weight: bold;
}

.NavBar {
    background-color: turquoise;
    color: aliceblue;
    position: relative;
    text-align: right;
}

a {
    display: inline-block;
    position: relative;
    padding: 0.5em 0.5em;
}

section {
    display: inline-block;
    margin-right: 1em; 
}

.NavBar-logo {
    left: 1em;
    position: absolute;
}