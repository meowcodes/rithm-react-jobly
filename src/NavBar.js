import React, {Component} from 'react';
import {NavLink} from 'react-router-dom';
import './NavBar.css'

/**For rendering links to all main pages 
 * everywhere in the app
 * changes based off whether user is/is not logged in
 *  */
class NavBar extends Component {

	render() {
		return ( 
			<nav className="NavBar" >
				<NavLink className="NavBar-logo" exact to="/">Jobly</NavLink> 
				{ this.props.currUser
					? <section>
						<NavLink to="/companies">Companies</NavLink>
						<NavLink exact to="/jobs">Jobs</NavLink>
						<NavLink exact to="/profile">Profile</NavLink>
						<a onClick={ this.props.triggerLogout } href="/">Log Out</a>
					</section>
					: <section>
						<NavLink exact to="/login">Log In</NavLink>
						<NavLink exact to="/register">Register</NavLink>
					</section>
				}
			</nav>
		);
	}
}

export default NavBar;