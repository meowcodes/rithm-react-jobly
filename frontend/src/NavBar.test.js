import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { MemoryRouter } from 'react-router-dom';
import NavBar from './NavBar';


// full mount
it('mounts without crashing', function() {
  mount(
    <MemoryRouter>
      <NavBar />
    </MemoryRouter>
  );
});

// snapshot test
it('matches snapshot', function() {
  let wrapper = shallow(<NavBar />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
