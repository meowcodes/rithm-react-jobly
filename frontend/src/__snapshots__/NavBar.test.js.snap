// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="NavBar"
>
  <NavLink
    className="NavBar-logo"
    exact={true}
    to="/"
  >
    Jobly
  </NavLink>
  <section>
    <NavLink
      exact={true}
      to="/login"
    >
      Log In
    </NavLink>
    <NavLink
      exact={true}
      to="/register"
    >
      Register
    </NavLink>
  </section>
</nav>
`;
